adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -507,6 +562,13 @@ private void doUploadMessage(final RoutingContext ctx, final String tenant, fina
             } else {
 
                 final Device authenticatedDevice = getAuthenticatedDevice(ctx);
+                Optional.ofNullable((Span) ctx.get(TracingHandler.CURRENT_SPAN)).map(span -> {
+                    span.setOperationName("upload " + endpointName);
+                    TracingHelper.TAG_TLS.set(span, ctx.request().isSSL());
+                    TracingHelper.TAG_AUTHENTICATED.set(span, authenticatedDevice != null);
+                    return span.context();
+                }).orElse(null);
```

Comment:
I would have expected an assignment to a `SpanContext` variable here - otherwise the `.orElse(null)` at the end is superfluous. And in that case a simple `if (span != null)` block would look more readable imho (also as `return span.context()` is ignored).

Reply:
will be assigned to a local variable in one of the next steps. Then it will all make sense ...